core/mgmt/v1beta/metric.proto

```diff
@@ -68,110 +55,119 @@ message PipelineTriggerTableRecord {
-  // Aggregation window in nanoseconds.
-  int32 aggregation_window = 1;
-  // Filter can hold an [AIP-160](https://google.aip.dev/160)-compliant filter
-  // expression.
-  // - Example: `create_time>timestamp("2000-06-19T23:31:08.657Z")`.
-  optional string filter = 2 [(google.api.field_behavior) = OPTIONAL];
+  // 1 is reserved for the aggregation window in nanoseconds. This is
+  // deprecated in favour of an aggregation window string that represents a
+  // duration.
+  reserved 1;
+  // 2 is reserved for the filter. For now, this endpoint won't allow filtering
+  // but in the future we might implement a filter to show the trigger count of
+  // only certain pipelines and to group by the pipeline ID.
+  reserved 2;
+
+  // The ID of the pipeline trigger requester.
+  // Format: `{[users|organizations]}/{id}`.
+  string requester = 3 [(google.api.field_behavior) = REQUIRED];
+  // Aggregation window. The value is a positive duration string, i.e. a
+  // sequence of decimal numbers, each with optional fraction and a unit
+  // suffix, such as "300ms", "1.5h" or "2h45m".
+  // The minimum (and default) window is 1h.
+  optional string aggregation_window = 4;
+  // Beginning of the time range from which the records will be fetched.
+  // The default value is the current timestamp.
+  optional google.protobuf.Timestamp start = 5;
+  // End of the time range from which the records will be fetched.
+  // The default value is the beginning of the current day, in UTC.
+  optional google.protobuf.Timestamp stop = 6;
 }
 
 // ListPipelineTriggerChartRecordsResponse contains a list of pipeline trigger
 // chart records.
 message ListPipelineTriggerChartRecordsResponse {
-  // A list of pipeline trigger records.
+  // Pipeline trigger counts. Until we allow filtering or grouping by fields
+  // like pipeline ID, this list will contain only one element with the
+  // timeline of trigger counts for a given requester, regardless the pipeline
+  // ID, trigger mode, final status or other fields.
   repeated PipelineTriggerChartRecord pipeline_trigger_chart_records = 1;
 }
 
-// CreditConsumptionChartRecord contains credit consumption metrics, aggregated
-// by owner and time frame.
+// CreditConsumptionChartRecord represents a timeline of Instill Credit
+// consumption. It contains a collection of (timestamp, amount) pairs that
+// represent the total credit consumption in a given time bucket.
 message CreditConsumptionChartRecord {
   // Credit owner ID, e.g. `users/chef-wombat`.
-  string credit_owner = 1;
+  string credit_owner = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
   // Time buckets.
   repeated google.protobuf.Timestamp time_buckets = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
   // Total credit consumed in each time bucket.
   repeated float amount = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Credit consumption source (e.g. "pipeline", "model").
+  string source = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
ℹ️ If in the future we filter / group results by pipeline ID, model ID, version, release, etc., we'll add some `optional` fields here.